core/src/main/java/org/axonframework/eventsourcing/SpringAggregateSnapshotter.java

```diff
@@ -73,9 +74,34 @@ public void setTransactionDefinition(TransactionDefinition transactionDefinition
         this.transactionDefinition = transactionDefinition;
     }
 
-    @SuppressWarnings({"unchecked"})
+    /**
+     * Optionally sets the aggregate factories to use. By default, this implementation will auto detect available
+     * factories from the application context. Configuring them using this method will prevent auto detection.
+     *
+     * @param aggregateFactories The list of aggregate factories creating the aggregates to store.
+     */
     @Override
-    public void afterPropertiesSet() throws Exception {
+    public void setAggregateFactories(List<AggregateFactory<?>> aggregateFactories) {
+        this.autoDetectAggregateFactories = false;
+        super.setAggregateFactories(aggregateFactories);
+    }
+
+    @Override
+    public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+        this.applicationContext = applicationContext;
+    }
+
+    @Override
+    public boolean isAutoStartup() {
+        return true;
+    }
+
+    @Override
+    public void stop(Runnable runnable) {
```

Comment:
You need to call runnable.run() here. And probably also want to set running to false (also in stop() )